java-checks/src/test/java/org/sonar/java/checks/sustainability/AndroidExactAlarmCheckTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.sustainability;
+
+import org.junit.jupiter.api.Test;
+import org.sonar.java.checks.methods.AbstractMethodDetection;
+import org.sonar.java.checks.security.AndroidBroadcastingCheck;
+import org.sonar.java.checks.verifier.CheckVerifier;
+
+import static org.sonar.java.checks.verifier.TestUtils.mainCodeSourcesPath;
+
+
+  import org.junit.jupiter.api.Test;
+  import org.sonar.java.checks.verifier.CheckVerifier;
+import org.sonar.plugins.java.api.InputFileScannerContext;
+import org.sonar.plugins.java.api.semantic.MethodMatchers;
+
+import static org.sonar.java.checks.verifier.TestUtils.mainCodeSourcesPath;
+
+class AndroidExactAlarmCheckTest {
+
+  @Test
+  void test() {
```

Comment:
Since we are cheating in our test sources for Android rules, it may be worth adding an extra test to test that nothing is raised without semantics. It should work out of the box, you can use the same file you have anyway. Not a huge point, happy to discuss.

Reply:
Added check for android context in report function + second test `withAndroidContext(false)`.